packages/quicktype-core/src/language/Dart/DartRenderer.ts

```diff
@@ -469,7 +469,7 @@ export class DartRenderer extends ConvenienceRenderer {
                     defined(this._enumValues.get(enumType)),
                     ".map[",
                     dynamic,
-                    this._options.nullSafety ? "]!" : "]",
+                    (this._options.nullSafety && !isNullable) ? "]!" : "]"
```

Comment:
```suggestion
                     (this._options.nullSafety && !enumType.isNullable && !isNullable)? "]!" : "]",
```

In my tests, this is the code that worked